CodeContinue: LLM powered code completion tool

kXborg opened the pull request:
<!--
Please have patience, we usually need a few weeks to get around to reviewing your submission. 
To help us do so, please provide some information about your package:
-->

- [x] I'm the package's author and/or maintainer.
- [x] I have have read [the docs][1].
- [x] I have tagged a release with a [semver][2] version number.
- [x] My package repo has a description and a README describing what it's for and how to use it.
- [x] My package doesn't add context menu entries. *
- [x] My package doesn't add key bindings. **
- [x] Any commands are available via the command palette.
- [x] Preferences and keybindings (if any) are listed in the menu and the command palette, and open in split view.
- [x] If my package is a syntax it doesn't also add a color scheme. ***
- [x] I use [.gitattributes][3] to exclude files from the package: images, test files, sublime-project/workspace.

[1]: https://docs.sublimetext.io/guide/package-control/submitting.html
[2]: https://semver.org
[3]: https://www.git-scm.com/docs/gitattributes#_export_ignore

My package is codeContinue. 
 - It's build to provide LLM powered code completion.
 - Just start with 1-2 line of comments and continue coding, it suggest next line of code instantly
 - Use any OpenAI compatible endpoint to serve a model and use the same here. I use Qwen2.5-Coder-1.5B-Instruct on a jetson orin nano from vLLM. It works decent.
 - The goal is to have control over the models we use, and keep data local.

Disclaimer: I am aware of potential key binding conflicts, however `Tab` key rarely will be creating issue. Also it is only triggered when suggested code phantom appears while coding. We have tested it rigorously, and we beleive it should not create issues. Having said that, the instructions have been added to change it in the first run -- in case any conflicts exists. I hope the package will not be declined. 

There are no packages like it in Package Control.

*)   If you do need a context menu, make sure the menu applies to the cursor
     context, and the commands are conditional. Space in this menu is limited!
**)  There aren't enough keys for all packages, so you risk overriding those
     of other packages. You can put commented out suggestions in a keymap file, 
     and/or explain how to create bindings in your README.
***) Syntaxes should work in any color scheme the user chooses.

For bonus points also consider how the review guidelines apply to your package:
https://docs.sublimetext.io/reference/package-control/reviewing.html

braver commented on repository/c.json:
```diff
@@ -2081,6 +2081,19 @@
+			"name": "CodeContinue",
+			"details": "https://github.com/kXborg/codeContinue",
+			"author": "Kukil Kashyap Borgohain",
+			"description": "AI Code Completion for Sublime Text",
```
```suggestion
```

You can leave this out, the GitHub repository description will be taken instead, making it easier for you to make changes.

kXborg commented:
@braver,
Thanks again. Made the following changes.
- Added Package Settings menu with Settings, Key Bindings, and Configure Wizard entries
- Added 2 new commands: `Preferences: CodeContinue Settings` and `Preferences: CodeContinue Key Bindings`
- Added `"debug": false and "api_key": ""` with documentation comments
- `log()` now only prints when debug setting is true

Please check and let me know.

FichteFoll commented:
> I'm actually not sure if `sublime.save_settings` writes to the User package automatically?

It does.

braver commented:
> Please check and let me know.

For keybindings it's usually preferable to open the user's global (ie. not package-specific) keybindings file on the right. You usually want to manage all your bindings in one place, because you've only got one keyboard and typically want to have a clear overview of what you've set to which key combo. [example](https://github.com/sublimelsp/LSP/blob/main/Main.sublime-menu#L301)

Also please revert the formatting of the repository file in this PR.
